### PR TITLE
feat(apps/interface): create my swap container

### PR DESCRIPTION
**Pull request by asaadam**
create my swap container

### CODE REVIEW

File: apps/interface/components/SwapHistoryCard/MySwapContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from "react";
+import { useAccount } from "wagmi";
+
+// Hooks
+import useFuseLeveragedTokenSwap from "@/hooks/useFuseLeveragedTokenSwap";
+
+// Sub-components
+import SwapHistoryCardTable from "./Table";
+import type { FuseLeveragedTokenSwap } from "@/utils/types";
```
Comment (pyk) [inline apps/interface/components/SwapHistoryCard/MySwapContainer.tsx:9]: Add utils to its own group

```ts
// Utils
import type { FuseLeveragedTokenSwap } from "@/utils/types";

// Sub-components
...
```

File: apps/interface/components/SwapHistoryCard/MySwapContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+interface MySwapHisotryContainerProps {
+    symbol: string;
+}
+
+const dummyData: FuseLeveragedTokenSwap = {
+    amountInUSD: "0",
+    timestamp: (new Date().getTime() / 1000).toString(),
```
Comment (pyk) [inline apps/interface/components/SwapHistoryCard/MySwapContainer.tsx:17]: just use dummy number here like `1659328398`

File: apps/interface/components/SwapHistoryCard/MySwapContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    tokenIn: {
+        symbol: "",
```
Comment (pyk) [inline apps/interface/components/SwapHistoryCard/MySwapContainer.tsx:19]: Use dummy symbol `BNBRISE`

File: apps/interface/components/SwapHistoryCard/MySwapContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    },
+    tokenOut: {
+        symbol: "",
```
Comment (pyk) [inline apps/interface/components/SwapHistoryCard/MySwapContainer.tsx:22]: Use dummy symbol `BUSD`

File: apps/interface/components/SwapHistoryCard/MySwapContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+export const MySwapHistoryContainer = (props: MySwapHisotryContainerProps) => {
+    const { symbol } = props;
+    const { address } = useAccount();
+    const { data, isLoaded } = useFuseLeveragedTokenSwap(symbol, address);
+    const [loadedData, setLoadedData] = useState<FuseLeveragedTokenSwap[]>(
+        Array.from(Array(5).keys()).map(() => dummyData)
+    );
+
+    useEffect(() => {
+        if (isLoaded && data) {
```
Comment (pyk) [inline apps/interface/components/SwapHistoryCard/MySwapContainer.tsx:38]: just use `if (data)`

File: apps/interface/components/SwapHistoryCard/Table.tsx [hunk 6]
```diff
@@ -76,15 +76,15 @@ export const SwapHistoryCardTable = (props: SwapHistoryCardTableProps) => {
                     width="100%"
                     alignItems="flex-start"
                 >
-                    {swaps.map((swap) => (
+                    {swaps.map((swap, index) => (
```
Comment (pyk) [inline apps/interface/components/SwapHistoryCard/Table.tsx:79]: @asaadam nice addition 👍

File: apps/interface/hooks/useFuseLeveragedTokenSwap.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import useSWR from "swr";
+import { FuseLeverageSwapsHistory } from "@risedle/types";
+
+import { fetcher } from "@/utils/fetcher";
+
+// Get list of markets using SWR
```
Comment (pyk) [inline apps/interface/hooks/useFuseLeveragedTokenSwap.ts:6]: typo here, it should be list of swaps

File: apps/interface/hooks/useFuseLeveragedTokenSwap.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+export function useFuseLeveragedTokenSwap(
```
Comment (pyk) [inline apps/interface/hooks/useFuseLeveragedTokenSwap.ts:7]: I think we need to rename it to `useFuseLeveragedTokenMySwap`

File: apps/interface/hooks/useFuseLeveragedTokenSwap.ts [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    symbol: string,
+    userAddress: string | undefined
+) {
+    let url = `https://api.risedle.com/v1/${symbol}/swaps/`;
+
+    if (userAddress) {
+        url += `${userAddress}/`;
+    }
```
Comment (pyk) [inline apps/interface/hooks/useFuseLeveragedTokenSwap.ts:15]: Set `url` to `null` if `userAddress` is null so swr doesn't perform the request when wallet is not connected.

Learn more here https://swr.vercel.app/docs/conditional-fetching

See example here https://github.com/risedle/monorepo/blob/733ae7b401ceed364151168d3237ed5bdcd54245/apps/interface/hooks/useFuseLeveragedTokenUserPosition.ts#L16-L22

File: apps/interface/hooks/useFuseLeveragedTokenSwap.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    const { data, error } = useSWR<FuseLeverageSwapsHistory, Error>(
+        url,
+        fetcher
+    );
+
+    return {
+        data: data,
+        isLoaded: data?.user !== undefined ? true : false,
```
Comment (pyk) [inline apps/interface/hooks/useFuseLeveragedTokenSwap.ts:23]: return loaded if data.user is not null or `userAddress` is null.

See example here:

https://github.com/risedle/monorepo/blob/733ae7b401ceed364151168d3237ed5bdcd54245/apps/interface/hooks/useFuseLeveragedTokenUserPosition.ts#L24-L28

So in stateless component, we need to handle state where `isLoaded = true` and `data=null` to display message such as "Wallet is not connected"

File: apps/interface/tests/components/MySwapHistoryContainer.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,27 @@
+import { render, screen } from "@testing-library/react";
+import * as wagmi from "wagmi";
+
+// import QuoteBalanceContainer from "@/components/SwapCard/QuoteBalanceContainer";
```
Comment (pyk) [inline apps/interface/tests/components/MySwapHistoryContainer.spec.tsx:4]: remove this line

File: apps/interface/tests/components/MySwapHistoryContainer.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+import MySwapHistoryContainer from "@/components/SwapHistoryCard/MySwapContainer";
+import renderApp from "../utils/renderApp";
+
+afterEach(() => {
+    // restore the spy created with spyOn
+    jest.restoreAllMocks();
+});
+
+describe("<QuoteBalanceContainer />", () => {
```
Comment (pyk) [inline apps/interface/tests/components/MySwapHistoryContainer.spec.tsx:13]: typo MySwapHistoryContainer

File: apps/interface/tests/hooks/useFuseLeveragedTokenSwap.spec.ts [hunk 13]
```diff
@@ -0,0 +1,49 @@
+import useFuseLeveragedTokenSwap from "@/hooks/useFuseLeveragedTokenSwap";
+import useFuseLeveragedTokenUserPosition from "@/hooks/useFuseLeveragedTokenUserPosition";
```
Comment (pyk) [inline apps/interface/tests/hooks/useFuseLeveragedTokenSwap.spec.ts:2]: remove unused imports

File: apps/interface/tests/hooks/useFuseLeveragedTokenSwap.spec.ts [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+import { renderHook } from "@testing-library/react";
+import * as SWR from "swr";
+
+describe("useFuseLeveragedTokenUserPosition", () => {
```
Comment (pyk) [inline apps/interface/tests/hooks/useFuseLeveragedTokenSwap.spec.ts:6]: typo

File: apps/interface/tests/hooks/useFuseLeveragedTokenSwap.spec.ts [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+        it("should return isLoaded=false and data=undefined", async () => {
+            const { result } = renderHook(() =>
+                useFuseLeveragedTokenSwap("test", undefined)
+            );
+            expect(result.current.isLoaded).toBe(false);
+            expect(result.current.data).toBe(undefined);
+        });
+    });
+
+    describe("Given only symbol and address", () => {
```
Comment (pyk) [inline apps/interface/tests/hooks/useFuseLeveragedTokenSwap.spec.ts:17]: typo: `Given symbol and address`

**Review by pyk — CHANGES_REQUESTED** [review]

**Comment by pyk** [comment]
@asaadam the build response from vercel is not helpful at all. Not sure what happen

```
Error [SyntaxError]: Unexpected token < in JSON at position 1480481
--
20:49:54.285 | at JSON.parse (<anonymous>)
20:49:54.285 | at Object.readConfigFile (/var/task/node_modules/@vercel/build-utils/dist/index.js:30773:29)
20:49:54.285 | at async Promise.all (index 1)
20:49:54.285 | at scanParentDirs (/var/task/node_modules/@vercel/build-utils/dist/index.js:31017:58)
20:49:54.285 | at getNodeVersion (/var/task/node_modules/@vercel/build-utils/dist/index.js:30975:29)
20:49:54.285 | at Object.build (/var/task/node_modules/@vercel/next/dist/index.js:42745:25)
20:49:54.285 | at doBuild (/var/task/node_modules/vercel/dist/index.js:239130:33)
20:49:54.285 | at main (/var/task/node_modules/vercel/dist/index.js:238956:16)
20:49:54.286 | at main (/var/task/node_modules/vercel/dist/index.js:246231:20)


```

File: apps/interface/hooks/useFuseLeveragedTokenSwap.ts (hunk 8 above)
Comment (asaadam) [reply apps/interface/hooks/useFuseLeveragedTokenSwap.ts:7]: should i rename filename too ?

File: apps/interface/hooks/useFuseLeveragedTokenMySwap.ts [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+
+import { fetcher } from "@/utils/fetcher";
+
+// Get list of markets using SWR
+export function useFuseLeveragedTokenMySwap(
+    symbol: string,
+    userAddress: string | undefined
+) {
+    const url = userAddress
+        ? `https://api.risedle.com/v1/${symbol}/swaps/${userAddress}`
```
Comment (pyk) [inline apps/interface/hooks/useFuseLeveragedTokenMySwap.ts:12]: this should be `https://api.risedle.com/v1/${chainId}/flts/${symbol}/swaps/${userAddress}`

**Review by pyk — CHANGES_REQUESTED** [review]

**Comment by pyk** [comment]
@asaadam What's status of this pull request? we need to wait for the design right?

lets convert this to draft then

**Comment by asaadam** [comment]
> @asaadam What's status of this pull request? we need to wait for the design right?
> 
> lets convert this to draft then

yap we still need to wait for the design ser

okay i will convert it to draft

**Pull request closed without merging**